courses-app/src/app/app-routing.module.ts

```diff
@@ -1,10 +1,34 @@
 import { NgModule } from '@angular/core';
 import { RouterModule, Routes } from '@angular/router';
+import { AuthorizedGuard } from './auth/guards/authorized.guard';
+import { NotAuthorizedGuard } from './auth/guards/not-authorized.guard';
 
-const routes: Routes = [];
+const routes: Routes = [
+  {
+    path: 'login',
+    loadChildren: () => import('./features/login/login.module').then(module => module.LoginModule),
+    canActivate: [NotAuthorizedGuard]
+  },
+  {
+    path: 'registration',
+    loadChildren: () => import('./features/registration/registration.module').then(module => module.RegistrationModule),
+    canActivate: [NotAuthorizedGuard]
+  },
+  {
+    path: 'courses',
+    loadChildren: () => import('./features/courses/courses.module').then(module => module.CoursesModule),
+    canLoad: [AuthorizedGuard]
+  },
+  {
+    path: '',
+    redirectTo: '/courses',
+    pathMatch: 'full'
+  },
+];
 
 @NgModule({
   imports: [RouterModule.forRoot(routes)],
+  providers: [AuthorizedGuard, NotAuthorizedGuard],
```

Comment:
it would be better to use @Injectable like you did. It doesn't makes much sense to inject it twice to root injector